File: 프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    people = deque(sorted(people))  # 오름차순 정렬
    ans = 0
    
    while people:
        n = len(people)
        if n > 1 and people[0] + people[-1] <= limit:
            people.popleft()
        
        people.pop()
        ans += 1
        
    return ans

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
    var ans = 0;
    people.sort((a, b) => b - a); // 내림차순
    for (let i = 0; i < people.length; i++) {
        ans += 1;
        if (people[i] + people[people.length - 1] <= limit) {
            people.pop();
        }
    }
    return ans;
}